cuqi/implicitprior/_restorator.py

```diff
@@ -54,9 +57,18 @@ def restore(self, x, restoration_strength):
             restorator is a denoiser, this parameter might correspond to the
             noise level.
         """
-        solution, info = self.restorator(x, restoration_strength=restoration_strength,
+        restorator_return = self.restorator(x, restoration_strength=restoration_strength,
                                          **self.restorator_kwargs)
-        self.info = info
+
+        if type(restorator_return) == tuple and len(restorator_return) == 2:
+            solution, self.info = restorator_return
+        else:
+            raise ValueError("Wrong return type from the restorator. "+ 
+                             "Please ensure that the restorator returns a two-element tuple with the "+
+                             "restored solution as the first element and additional information about the "+
+                             "restoration as the second element. As to the second element, it can be of any type, "+
+                             "including `None` in case there is no particular information.")
+
```

Comment:
```suggestion
            raise ValueError("Unsupported return type from the user-supplied restorator function. "+ 
                             "Please ensure that the restorator function returns a two-element tuple with the "+
                             "restored solution as the first element and additional information about the "+
                             "restoration as the second element. The second element can be of any type, "+
                             "including `None` in case there is no particular information.")

```